src/blog/2024/09/node-red-if-else-logic-guide.md

```diff
@@ -0,0 +1,216 @@
+
+The [Switch](/node-red/core-nodes/switch/) node in Node-RED is used for routing messages based on specific conditions, offering a straightforward, low-code approach to implementing conditional logic in your flows. The Switch node allows you to set up rules using a visual interface, making it ideal for users who prefer a more intuitive method for handling conditions. However, it’s important to note that the Switch node represents a different, independent concept known as the "switch statement." While it serves a similar purpose to If-Else logic by building conditional flows, it operates under its own programming paradigm.
+
+To demonstrate the Switch node, we'll set up a flow to make decisions based on the temperature value. We will route messages through different outputs based on temperature thresholds.
+
+1. Drag the inject node onto the canvas and set the `msg.payload` to `$random() * 100` as JSONata expression; this inject node will simulate a temperature sensor by generating a random number.
+2. Drag a Switch node onto the canvas. Double-click on it and set Property to `msg.payload`.
+3. To add rules, click the + Add button at the bottom left of the configuration panel. You will see a prompt to select the condition and a prompt to enter the value to compare with. Add the following four rules and set it for "checking all rules":
+
+    - Rule 1: `msg.payload > 30` 
+    - Rule 2: `msg.payload <= 30` 
+    - Rule 3: `msg.payload <= 20` 
+    - Rule 4: `msg.payload <= 10`
+
+4. Now drag another Switch node and connect its input to the output of Switch nodes 2 and 3. We are adding a second switch node because we need to route messages based on ranges. A single Switch node doesn’t allow multiple checks in one rule, so we need to use another Switch node to route the temperature based on ranges. Add the following rules and set it for "stopping after the first match":
+
+    - Rule 1: `msg.payload > 20` 
+    - Rule 2: `msg.payload > 10` 
+
+5. Now drag the Debug nodes and connect them to the Switch nodes' outputs according to our example. For messages greater than 30, connect the Debug node to the first output of the first Switch node. For the range between 30 to 20, connect the Debug node to the first output of the second Switch node. For the range between 20 to 10, connect the Debug node to the second output of the second Switch node. Finally, for messages less than 10, connect the Debug node to the fourth output of the first Switch node.
+6. Deploy the flow by clicking the "Deploy" button in the top-right corner of the Node-RED editor.
+7. Once deployed, click the button on the Inject node to trigger it. The Debug nodes will show the routed messages based on the temperature value.
+
+Notice how messages are routed through different outputs based on the temperature value. Now, you may ask how to update the message payload based on a condition. For that, you will need to use the Change node or the Function node.
+
+```mermaid
```

Comment:
@Yndira-E Can you check why these aren't rendered? I think the library needs loading?

Reply:
Maybe it is not rendering in the Netlify preview.

Reply:
You were right, ZJ. I've added the Mermaid script to the posts template and now the diagrams are rendering correctly.